### PR TITLE
[cluster-test] Remove dependency on external dependency : retry

### PR DESCRIPTION
**Pull request by ankushagarwal**
## Summary

Get rid of external dependency `retry`. Add a simple implementation of retry under `common/util`.

## Test Plan

Added unit tests

### CODE REVIEW

File: common/util/src/retry.rs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    I: IntoIterator<Item = Duration>,
+    O: FnMut() -> Result<T, E>,
+{
+    let mut iterator = iterable.into_iter();
+    loop {
+        match operation() {
+            Ok(value) => return Ok(value),
+            Err(err) => {
+                if let Some(delay) = iterator.next() {
+                    sleep(delay);
```
Comment (andll) [inline common/util/src/retry.rs]: nit: I think we usually don't import functions, e.g. use `thread::sleep` instead of `sleep`
@bors-libra delegate+

**Comment by andll** [comment]
cc @bmwill

File: common/util/src/retry.rs (hunk 1 above)
Comment (ankushagarwal) [reply common/util/src/retry.rs]: Done

**Comment by bmwill** [comment]
Looks good to me, only nit would be that the crate name "util" isn't too descriptive and we already have a "tools" crate that exists

**Pull request closed without merging**